### PR TITLE
xpad: simplify reporting xbox one home button

### PR DESCRIPTION
**Pull request by dantob**

### CODE REVIEW

File: xpad.c [hunk 1]
```diff
@@ -603,14 +603,28 @@ static void xpad360w_process_packet(struct usb_xpad *xpad, u16 cmd, unsigned cha
...
  */
-static void xpadone_process_buttons(struct usb_xpad *xpad,
-				struct input_dev *dev,
-				unsigned char *data)
+static void xpadone_process_packet(struct usb_xpad *xpad, u16 cmd, unsigned char *data)
 {
+	struct input_dev *dev = xpad->dev;
+
+	/* the xbox button has its own special report */
+	if (data[0] == 0X07) {
```
Comment (paroj) [inline xpad.c]: I actually liked the switch as it was before more

**Comment by paroj** [comment]
this just shuffles code around, and does not actually fix anything, right?

**Comment by dantob** [comment]
Yeah should be no changes. I was just trying to make it more similar to the other device types. xbox one was the one device that has a ####_process_buttons routine.

**Comment by paroj** [comment]
merged manually

**Pull request closed without merging**